tests/Microsoft.Identity.Test.Unit.net45/CoreTests/WsTrustTests/WsTrustEndpointTests.cs

```diff
@@ -164,6 +165,39 @@ public void TestWsTrustEndpointUsernamePasswordWsTrust13()
             Assert.AreEqual(expectedMessage, requestMessage);
         }
 
+        [TestMethod]
+        [Description("Assert client info is included in userrealm discovery header")]
+        public void UserrealmDiscoveryHeadersTest()
+        {
+            string endpoint = "https://login.microsoftonline.com/common/userrealm/";
+
+            using (var harness = new MockHttpAndServiceBundle())
+            {
+                var requestContext = RequestContext.CreateForTest(harness.ServiceBundle);
+                var userrealmEndpoint = harness.ServiceBundle.WsTrustWebRequestManager.CreateUserrealmEndpoint(endpoint, _username, requestContext);
+
+                Assert.IsNotNull(userrealmEndpoint);
+
+                string[] pathSegments = userrealmEndpoint.Uri.AbsoluteUri.Substring(0).Split(
```

Comment:
The Uri class already does the parsing for you - is there a reason you can't use it?